### DISCUSSION TITLE
Make the `.server` and `.client` folder names configurable

### DISCUSSION THREAD

**Discussion by heivo** [remix-run/remix#10035] [Proposals]
I like the possibility to move all [server-only](https://remix.run/docs/en/main/file-conventions/-server) or [client-only](https://remix.run/docs/en/main/file-conventions/-client) modules inside a dedicated folder, however the leading dot in the default folder names (`.server` and `.client`) has some drawbacks:

* those folders are hidden by default in Linux
* the tsconfig.json include config has to be adjusted (e.g. `{ "include": ["app/.server/**/*.ts"] }`)
* eslint has to be configured to not exclude files within those folders (e.g. `{ "ignorePatterns": ['!app/.server/'] }`)

It would be nice if the name of those folders were configurable, for example:
```ts
export default defineConfig({
  plugins: [
    remix({
      serverModulesDirectory: 'server',
      clientModulesDirectory: 'client',
    }),
  ],
})
```

**Comment by khera** [comment]
It would be great if it would work like file names, where the `.server` can be suffix of the folder name rather than exactly that name.

**Comment by ryanxcharles** [comment]
yes, i agree. it would be great if the folders could be configured. i use them extensively because i think this is easier to use than putting ".server" and ".client" in every filename. however, sometimes i forget to do "ls -a" and confuse myself when i can't see all the folders.

**Comment by sergiodxa** [comment]
You can use `vite-env-only` plugin to configure in your Vite config what folders or files are server-only and client-only, even use the macro to mark specific exports as server-only or client-only.

**Reply by heivo** [reply]
Thank you, didn't know that this exists, will try it out.

Nevertheless it would be nice if RR Framework supported this out of the box (or at least had a hint in the docs how to configure it).

**Reply by sergiodxa** [reply]
Remix docs mention it here https://remix.run/docs/en/main/discussion/server-vs-client#vite-env-only, most Remix docs apply to RR, and RR docs are being improved by the team, I recommend you to check Remix docs too.
